compiler/rustc_lint/src/builtin.rs

```diff
@@ -2409,8 +2409,34 @@ impl<'tcx> LateLintPass<'tcx> for InvalidValue {
         }
 
         /// Information about why a type cannot be initialized this way.
-        /// Contains an error message and optionally a span to point at.
-        type InitError = (String, Option<Span>);
+        struct InitError {
+            message: String,
+            /// Spans from struct fields and similar can be obtained from just the type.
+            span: Option<Span>,
```

Comment:
I'm confused by this comment. What is the span that is being stored here?

Or is there a "that" missing?
```suggestion
            /// Spans from struct fields and similar that can be obtained from just the type.
            span: Option<Span>,
```

Reply:
this review is still relevant

Reply:
this review is still relevant